README.md

```diff
@@ -9,3 +9,61 @@ See [Declarative String Processing Overview][decl-string]
 ## Requirements
 
 - [Swift Trunk Development Snapshot](https://www.swift.org/download/#snapshots) DEVELOPMENT-SNAPSHOT-2022-02-03 or later.
+
+## Integration with Swift
+
+### Specially integrated modules
+
+`_MatchingEngine`, `_CUnicode` and `_StringProcessing` are specially integrated modules that are built as part of apple/swift.
+
+Specifically, `_MatchingEngine` contains the parser for regular expression literals and is built both as part of the compiler and as a core library. `_CUnicode` and `_StringProcessing` are built together as a core library named `_StringProcessing`.
+
+| Module              | Swift toolchain component                                                            |
+| ------------------- | ------------------------------------------------------------------------------------ |
+| `_MatchingEngine`   | `SwiftCompilerSources/Sources/ExperimentalRegex` and `stdlib/public/_MatchingEngine` |
+| `_CUnicode`         | `stdlib/public/_StringProcessing`                                                    |
+| `_StringProcessing` | `stdlib/public/_StringProcessing`                                                    |
+
+### Branches
+
+`main` and `release/` branches correspond to branches in apple/swift of the same name.  They are expected to build together and pass all tests.  
```

Comment:
Part of the reason we're taking on significant integration risk here, instead of source drops, is to help develop better ways to decrease lock and coupling between repos for the future. This change runs directly counter to that goal, pulling in all the preexisting tight coupling into this new model. What's the justification?

Reply:
Reverted to using swift/main after discussions.